plugins/catalog/src/components/CatalogPage/CatalogPage.tsx

```diff
@@ -46,39 +34,31 @@ export type CatalogPageProps = {
 };
 
 export const CatalogPage = ({
-  initiallySelectedFilter = 'owned',
   columns,
   actions,
-}: CatalogPageProps) => (
-  <CatalogLayout>
-    <Content>
-      <ContentHeader title="Components">
-        <CreateComponentButton />
-        <SupportButton>All your software catalog entities</SupportButton>
-      </ContentHeader>
-      <Grid container spacing={2}>
-        <EntityListProvider>
-          <Grid item sm={12} lg={2} alignContent="flex-start">
-            <Grid container>
-              <Grid item xs={12} sm={4} lg={12}>
-                <EntityKindPicker initialFilter="component" hidden />
-                <EntityTypePicker />
-              </Grid>
-              <Grid item xs={12} sm={4} lg={12}>
-                <UserListPicker initialFilter={initiallySelectedFilter} />
-              </Grid>
-              <Grid item xs={12} sm={4} lg={12}>
-                <EntityOwnerPicker />
-                <EntityLifecyclePicker />
-                <EntityTagPicker />
-              </Grid>
-            </Grid>
-          </Grid>
-          <Grid item xs={12} sm={12} lg={10}>
-            <CatalogTable columns={columns} actions={actions} />
-          </Grid>
-        </EntityListProvider>
-      </Grid>
-    </Content>
-  </CatalogLayout>
-);
+  initiallySelectedFilter = 'owned',
+}: CatalogPageProps) => {
+  const orgName =
+    useApi(configApiRef).getOptionalString('organization.name') ?? 'Backstage';
+  const createComponentLink = useRouteRef(createComponentRouteRef);
+
+  return (
+    <EntityListProvider>
+      <TablePage
```

Comment:
I appreciate the goal of this contribution, but I think the components introduced here are going the wrong direction for [catalog customization](https://backstage.io/docs/features/software-catalog/catalog-customization) which aims to make this page easy to pick apart and customize.

The `CatalogPage` JSX structure, according to that goal, should be very easy to read and understand, with minimal layers in between what you see on the screen and the page structure in code. The addition of `TablePage` and `CatalogFilter` here add undesirable layers, in my view. Even the `Grid` components from #6342 are unfortunate since it grows the page structure that was purposefully minimized in #5643.

Apologies that this wasn't clear, we should probably make an ADR around this assuming that we want to keep the intentions of #5643, at least until there's a more fleshed out approach to customization and composability in Backstage as a whole.

Reply:
I do think however, that we shouldn't sacrifice behaviour just because it's a nice JSX tree. It would be good if we can find some sort of middle ground. Maybe achieving the end result by using existing JSX elements or something?

Reply:
@tudi2d @benjdlambert WDYT about something like this?

```tsx
<CatalogLayout>
  <Content>
    <ContentHeader title="Components">
      <CreateComponentButton />
      <SupportButton>All your software catalog entities</SupportButton>
    </ContentHeader>
    <EntityListProvider>
      <FilteredTableLayout>
        <FilterContainer>
          <EntityKindPicker initialFilter="component" hidden />
          <EntityTypePicker />
          <UserListPicker initialFilter={initiallySelectedFilter} />
          <EntityOwnerPicker />
          <EntityLifecyclePicker />
          <EntityTagPicker />
        </FilterContainer>
        <CatalogTable columns={columns} actions={actions} />
      </FilteredTableLayout>
    </EntityListProvider>
  </Content>
</CatalogLayout>
```

This should be enough structure to handle filters on the side or stacked, and the collapse behavior. This loses the filter grouping from #6342 at medium screen sizes. I'm slightly worried about that grouping, it would be easy for someone popping in new filters to fumble the page layout on smaller resolutions without realizing it.

I think it could be better to switch to filter button that opens a drawer at smaller resolutions 🤔 Then the vertical layout is consistent, but still visible on the side when possible. This could be handled inside `FilterContainer`.

![filter-drawer](https://user-images.githubusercontent.com/556258/125342114-ad842580-e311-11eb-9186-d0cc3e4f2211.gif)

Reply:
So I gave it a try:

![filter_weird](https://user-images.githubusercontent.com/8904624/125504493-df9aaa56-4d6b-4512-8d3d-007bd966c47f.gif)

I did not built it the "Etsy"-way with a full-page drawer, as it would overlap with the sidebar making it impossible to navigate. Therefor I had to add a few hacks to the Drawer. It might look nice in the Demo-GIF, but it is really a workaround. The MUI Drawer is not meant for this & the site feels slow, when opening closing the drawer. As a site effect for some reason unmounting the FIlters causes the Catalog Entities to be unloaded.

All in all it's weird & I would like to mention, that this PR is now a Draft (can't change it on GitHub unfortunately) showing the possibility. There might be some possible tweaks making it more stable, but it will always be pretty hacky

-----

I agree, that the CatalogPage should have a better overview than in the initial PR. Nevertheless I also agree with @benjdlambert that we should not sacrifice usability for it. My suggestion would be to do smth. like this:

Reply:
It's a bit hidden, but you can covert the PR to a draft at the top right:

![image](https://user-images.githubusercontent.com/648527/125507201-78252140-8398-4064-b7d2-21acffde6dc7.png)

Reply:
I like the behavior! I would bring the filter button down to just above the table, and make it a bit more obvious with a text label. The catalog entity unloading side effect definitely sounds unexpected and warrants some troubleshooting.

I don't think a drawer covering up the nav is a non-starter, necessarily. It can be mildly inconvenient to have nav blocked, of course - but less so as the result of a purposeful action from a user, who is on a knowingly-constrained small screen. I don't personally get offended on mobile when a settings-type screen gives me maximum real estate by covering up some things I probably don't need in that moment.

The Material Design site has a modal-style drawer entering from the right in one example from the [filter-chips section](https://material.io/components/chips#filter-chips). Etsy's example covers up the nav as well; not that I put a ton of stock in their design, it was just an example that got me thinking.

Reply:
Ok - now I'm more happy with it. Good for now I think. Thought it would be worse with the overlapping Drawer, but as the screen size gets smaller the visible space is limited anyway, such that navigating & filtering at the same time is not really needed. Ready to review :)

| Catalog | API Explorer |
|:----:|:----:|
| ![filters_drawer](https://user-images.githubusercontent.com/8904624/125613692-12447078-1342-4775-9418-435a617b0a9a.gif) | ![filters_drawer_api](https://user-images.githubusercontent.com/8904624/125617683-e80697cf-2806-4f82-ba88-5f6ba4b7d68a.gif) |

Reply:
Btw, the Backstage Figma puts the toggle button into the table header: 
![image](https://user-images.githubusercontent.com/648527/125628492-54ae6b36-a94f-4424-b92f-a69fee9c4640.png)

https://www.figma.com/community/file/850673348101741100?preview=fullscreen

Reply:
Thanks @Fox32 - wrote with Tim yesterday on Slack about the component. It was used for the [API Explorer before being switched over for consistency](https://github.com/backstage/backstage/pull/5984). Not sure if things changed, but I understood it the way, that I should not use it here.

Reply:
@Fox32 Yeah, this is essentially showing the filters option for the `Table` component whereas these filters are a bit more broad beyond just `Table` and are also used on the scaffolder page that has a list of cards, for example. I'm trying to avoid tying the general filter/hook mechanism to any particular UI of the results display.

I was pondering with @tudi2d if we should remove the `Table` filter option altogether, it's confusing to have two approaches and will invite UI inconsistency.